### PR TITLE
tidy: add modernize-use-default-member-init

### PR DESCRIPTION
**Pull request by fanquake**
Refactor and then enable [`modernize-use-default-member-init`](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-default-member-init.html) in our `clang-tidy` job.

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1969,9 +1969,9 @@ static std::atomic<bool> g_should_abort_scan;
 class CoinsViewScanReserver
 {
 private:
-    bool m_could_reserve;
+    bool m_could_reserve{false};
 public:
-    explicit CoinsViewScanReserver() : m_could_reserve(false) {}
+    explicit CoinsViewScanReserver() {}
```
Comment (PastaPastaPasta) [inline src/rpc/blockchain.cpp:1974]:
```suggestion
    explicit CoinsViewScanReserver() = default;
```

**Review by PastaPastaPasta — COMMENTED** [review]
while you are here, I think it'd make sense to use `= default`

Otherwise, concept ACK

**Comment by fanquake** [comment]
> while you are here, I think it'd make sense to use = default

Have done that now.

**Review by PastaPastaPasta — APPROVED** [review]
utACK, I have reviewed the diff, and all changes make sense. I would recommend changing the title to `refactor: clang-tidy enable/use modernize-use-default-member-init and modernize-use-equals-default`

**Comment by MarcoFalke** [comment]
ACK modernize-use-default-member-init

not sure if modernize-use-equals-default provides any value for us, but I guess it doesn't hurt either.

**Review by 8498549767 — COMMENTED** [review]
Exelente

**Pull request merged**